code/modules/client/medal_persistence.dm

```diff
@@ -0,0 +1,267 @@
+	if(!query.warn_execute())
+		stack_trace("Failed to reset medals for ckey [ckey]")
+		qdel(query)
+		return FALSE
+	qdel(query)
+
+	var/any_medal_failed_to_save = FALSE
+	for(var/real_name in medals_by_real_name)
+		for(var/datum/persistent_medal_info/medal as anything in medals_by_real_name[real_name])
+			var/datum/db_query/query = SSdbcore.NewQuery({"INSERT INTO [format_table_name("medals")] (
+				ckey,
+				issued_to_real_name,
+				issued_to_rank,
+				issued_by_real_name,
+				issued_by_rank,
+				medal_typepath,
+				medal_citation,
+				issued_at,
+				is_posthumous
+			) VALUES (
+				:ckey,
+				:issued_to_real_name,
+				:issued_to_rank,
+				:issued_by_real_name,
+				:issued_by_rank,
+				:medal_typepath,
+				:medal_citation,
+				:issued_at,
+				:is_posthumous
+			)"}, list(
+				"ckey" = ckey,
+				"issued_to_real_name" = medal.issued_to_real_name,
+				"issued_to_rank" = medal.issued_to_rank,
+				"issued_by_real_name" = medal.issued_by_real_name,
+				"issued_by_rank" = medal.issued_by_rank,
+				"medal_typepath" = medal.medal_typepath,
+				"medal_citation" = medal.medal_citation,
+				"issued_at" = medal.issued_at,
+				"is_posthumous" = medal.is_posthumous,
+			))
+			if(!query.warn_execute())
+				any_medal_failed_to_save = TRUE
+				stack_trace("Failed to save medal for ckey [ckey]")
+				qdel(query)
+				continue
+			qdel(query)
+	if(any_medal_failed_to_save)
+		to_chat(user, span_warning("Some of your medals failed to save!"))
+
+/**
+ * Give all medals to a player.
+ * - mob/living/target - The player to give the medals to.
+ */
+/datum/medal_persistence/proc/give_medals_to(mob/living/target)
+	if(target.ckey != ckey)
+		stack_trace("Attempted to give medals to a player who does not match the ckey of the medal persistence object!")
+		return
+
+	var/obj/item/storage/box/medal_box = new
+	medal_box.name = "medal box"
+	medal_box.desc = "A box containing medals."
+
+	for(var/datum/persistent_medal_info/medal as anything in medals_by_real_name[target.real_name])
+		medal.award_to(medal_box)
+
+	if(!target.put_in_hands(medal_box))
+		medal_box.forceMove(get_turf(target))
+		to_chat(target, span_notice("There was no room in your hands for your medal box, so it was dropped on the ground."))
+	to_chat(target, span_notice("Don't forget to protect your medals!"))
+
+/**
+ * A datum to store persistent medal information.
+ */
+/datum/persistent_medal_info
+	var/datum/medal_persistence/medal_persistence //! The medal persistence object this medal is associated with
+
+	var/issued_to_real_name //! The real name of the player who was issued the medal
+	var/issued_to_rank //! The rank of the player who was issued the medal
+
+	var/issued_by_real_name //! The real name of the player who issued the medal
+	var/issued_by_rank //! The rank of the player who issued the medal
+
+	var/obj/item/clothing/tie/medal/medal //! The medal object
+	var/obj/item/clothing/tie/medal/medal_typepath //! The typepath of the medal
```

Comment:
I dont think a typepath is the best idea since those get changed around a bunch and a refactor nuking everything is suboptimal

persistent UID on the medal type perhaps